superset/common/query_context.py

```diff
@@ -285,7 +286,11 @@ def get_df_payload(  # pylint: disable=too-many-statements
                 status = utils.QueryStatus.FAILED
                 stacktrace = utils.get_stacktrace()
 
-            if is_loaded and cache_key and cache and status != utils.QueryStatus.FAILED:
```

Comment:
`and cache` is not needed because of https://github.com/apache/incubator-superset/pull/9020